### PR TITLE
Adriver id system

### PR DESCRIPTION
**Pull request by m-oranskaya**
## Type of change

- [x] New ID Module

## Description of change

Added AdriverIdModule (receives cookie and sets it via Prebid). Added work with cookies to AdriverBidAdapter. The received values ​​are passed in the request data. Added tests for adriverIdModule and additional tests for adriverBidAdapter.

- Maintainer email: m.oranskaya@adriver.ru

### CODE REVIEW

**Comment by m-oranskaya** [comment]
#3583

**Comment by m-oranskaya** [comment]
prebid/prebid.github.io#3583

**Comment by m-oranskaya** [comment]
https://github.com/prebid/prebid.github.io/pull/3583

**Comment by m-oranskaya** [comment]
Do we need to perform some additional actions before review?

File: modules/adriverIdSystem.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+              try {
+                responseObj = JSON.parse(response).adrcid;
+              } catch (error) {
+                logError(error);
+              }
+              let now = new Date();
+              now.setTime(now.getTime() + 86400 * 1825 * 1000);
+              storage.setCookie('adrcid', responseObj, now.toUTCString(), 'Lax');
+              storage.setDataInLocalStorage('adrcid', responseObj);
+              storage.setCookie('adrcid_cd', new Date().getTime(), now.toUTCString(), 'Lax');
```
[the remark this answers is not in the document]
Comment (m-oranskaya) [reply modules/adriverIdSystem.js:66]: We don't need to set a cookie to exactly 5 years, but we'd like to set it to the maximum possible time, as there are many situations where a longer cookie lifetime is useful. We didn't find any restrictions in the documentation about it, so we chose it ourselves. If you have any restrictions, then tell us them and we will set the maximum allowed value.

[the remark this answers is not in the document]
Comment (m-oranskaya) [reply modules/adriverIdSystem.js:66]: Do we need to change the cookie lifetime and are there any other comments on our code?

**Review by jlukas79 — APPROVED** [review]

**Pull request merged**